cmd/podman/parse/net.go

```diff
@@ -82,45 +82,40 @@ func GetAllLabels(labelFile, inputLabels []string) (map[string]string, error) {
 		}
 	}
 	for _, label := range inputLabels {
-		split := strings.SplitN(label, "=", 2)
-		if split[0] == "" {
+		key, value, _ := strings.Cut(label, "=")
+		if key == "" {
 			return nil, fmt.Errorf("invalid label format: %q", label)
 		}
-		value := ""
-		if len(split) > 1 {
-			value = split[1]
-		}
-		labels[split[0]] = value
+		labels[key] = value
 	}
 	return labels, nil
 }
 
 func parseEnvOrLabel(env map[string]string, line, configType string) error {
-	data := strings.SplitN(line, "=", 2)
+	key, val, hasVal := strings.Cut(line, "=")
 
 	// catch invalid variables such as "=" or "=A"
-	if data[0] == "" {
+	if key == "" {
 		return fmt.Errorf("invalid environment variable: %q", line)
 	}
 
 	// trim the front of a variable, but nothing else
-	name := strings.TrimLeft(data[0], whiteSpaces)
+	name := strings.TrimLeft(key, whiteSpaces)
 	if strings.ContainsAny(name, whiteSpaces) {
 		return fmt.Errorf("name %q has white spaces, poorly formatted name", name)
 	}
 
-	if len(data) > 1 {
-		env[name] = data[1]
+	if hasVal {
+		env[name] = val
 	} else {
-		if strings.HasSuffix(name, "*") {
-			name = strings.TrimSuffix(name, "*")
+		if name, hasStar := strings.CutSuffix(name, "*"); hasStar {
 			for _, e := range os.Environ() {
-				part := strings.SplitN(e, "=", 2)
-				if len(part) < 2 {
+				envKey, envVal, hasEq := strings.Cut(e, "=")
+				if hasEq {
```

Comment:
This does not seem equal to the old code, the condition needs to be inverted, we want to skip if the env is not valid (no `=`).
The fact that no test seem to catch this worries me.

Reply:
Seems like there is duplication with pkg/env hare, this needs to be looked at (you don't have to do that though)

Reply:
Fixed inverted check